scene/gui/button.cpp

```diff
@@ -53,73 +53,39 @@ void Button::_queue_update_size_cache() {
 void Button::_update_theme_item_cache() {
 	Control::_update_theme_item_cache();
 
+	theme_cache.max_style_size = Vector2();
+	theme_cache.style_margin_left = 0;
+	theme_cache.style_margin_right = 0;
+	theme_cache.style_margin_top = 0;
+	theme_cache.style_margin_bottom = 0;
+
 	const bool rtl = is_layout_rtl();
 	if (rtl && has_theme_stylebox(SNAME("normal_mirrored"))) {
```

Comment:
Since we have theme cache here I wonder if we can replace this check with a check against:
```suggestion
	if (rtl && theme_cache.normal_mirrored.is_valid()) {
```
I'm not extremely well versed with the theme system but looking at the code it seems it should be equivalent

Reply:
Not sure if it's the same, see https://github.com/godotengine/godot/pull/84944#issuecomment-1813395632

Reply:
Got it! Though I realized (hence unresolving) that if they *aren't* equivalent this code will segfault, so even if it's not a 100% drop in it does handle potential crashes